src/kms/TenantSecurityClient.ts

```diff
@@ -215,4 +218,42 @@ export class TenantSecurityClient {
      */
     logSecurityEvent = (event: SecurityEvent, metadata: EventMetadata): Promise<void> =>
         SecurityEventApi.logSecurityEvent(this.tspDomain, this.apiKey, event, metadata).toPromise();
+
+    /**
+     * Deterministically encrypt the provided field using the tenant's current secret.
+     */
+    deterministicEncryptField = (field: DeterministicPlaintextField, metadata: FieldMetadata): Promise<DeterministicEncryptedField> =>
+        KmsApi.deriveKey(this.tspDomain, this.apiKey, [field.derivationPath], field.secretPath, metadata)
+            .flatMap((deriveResponse) => Crypto.deterministicEncryptField(field, deriveResponse.derivedKeys[field.derivationPath]))
+            .toPromise();
+
+    /**
+     * Decrypt the provided deterministically encrypted field.
+     */
+    deterministicDecryptField = (encryptedDoc: DeterministicEncryptedField, metadata: FieldMetadata): Promise<DeterministicPlaintextField> =>
+        KmsApi.deriveKey(this.tspDomain, this.apiKey, [encryptedDoc.derivationPath], encryptedDoc.secretPath, metadata)
+            .flatMap((deriveResponse) => Crypto.deterministicDecryptField(encryptedDoc, deriveResponse.derivedKeys[encryptedDoc.derivationPath]))
+            .toPromise();
+
+    /**
+     * Decrypt the provided deterministically encrypted field and re-encrypt it with the current tenant secret.
+     */
+    deterministicReencryptField = (encryptedDoc: DeterministicEncryptedField, metadata: FieldMetadata): Promise<DeterministicEncryptedField> =>
```

Comment:
Yeah, I think we probably should. This will end up being more like the deprecated `rekeyDocument` than the `rekeyEdek`, but I think that's okay

Reply:
Thinking about this more, I wonder if there will be some confusion over reusing the name/concept. `rekeyDocument` is for switching to a new tenantId, but `rekeyField` would be for switching the tenantSecretId (the tenantId _can't_ change). Do you have thoughts?

Reply:
It's true that could be a source of confusion. I don't know if it'd be one that would bite people or not...

Reply:
We decided on `rotateField`. Migration will ensure you're always on the current secret unless there's been a rotation event, so that should be the only time this is called. May want to clarify that in the documentation, but the new name might be enough with the existing doc.